azkaban-common/src/main/java/azkaban/executor/ExecutableNode.java

```diff
@@ -63,7 +63,7 @@
   private Set<String> outNodes = new HashSet<>();
   private Props inputProps;
   private Props outputProps;
-  private int attempt = 0;
+  private volatile int attempt = 0;
```

Comment:
Why not use AtomicInteger?

Reply:
Yeah I don't mind, both work.. Do you? :)

Reply:
In my view, synchronized (this) should be too aggressive, because it locks out the whole object?

Reply:
Ok, I can change it.

Reply:
I was basically aiming at a less intrusive change. You'll see soon how much more complex the change becomes with AtomicInteger.